### ISSUE TITLE
Style/MixinUsage behaves inconsistently for guard clause

### ISSUE THREAD

**Issue by yskkin** [rubocop-hq/rubocop#7186]
## Expected behavior

Both of `WebRoutes` and `DebugRoutes` lines behaves same against rubocop

## Actual behavior

`Style/MixinUsage` violation is reported only for `DebugRoutes`

```
config/routes.rb:11:5: C: Style/MixinUsage: extend is used at the top level. Use inside class or module.
    extend DebugRoutes unless Rails.env.production?
    ^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Lint this file:

```ruby
Rails.application.routes.draw do
  # snip

  constraints(subdomain: 'xxxxxx') do
    extend WebRoutes
    extend DebugRoutes unless Rails.env.production?
  end

  # snip
end
```

## RuboCop version

```
$0.72.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-linux)
```

**Comment by koic** [comment]
Thanks for the feedback. I opened a PR #7194.
